common/src/main/java/io/druid/math/expr/Function.java

```diff
@@ -242,6 +243,27 @@ protected ExprEval eval(double param)
     }
   }
 
+  class Div extends DoubleParamMath
+  {
+    @Override
+    public String name()
+    {
+      return "div";
+    }
+
+    @Override
+    protected ExprEval eval(final long x, final long y)
+    {
+      return ExprEval.of(x / y);
+    }
+
+    @Override
+    protected ExprEval eval(final double x, final double y)
+    {
+      return ExprEval.of((long) (x / y));
```

Comment:
I didn't realize that. If so the original (long) x / (long) y was probably better, but I'm not sure. Up to you